h/services/annotation_metadata.py

```diff
@@ -0,0 +1,51 @@
+import json
+
+from sqlalchemy import func, select
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, AnnotationMetadata, AnnotationSlim
+from h.security import decrypt_jwe_dict
+
+
+class AnnotationMetadataService:
+    def __init__(self, db: Session, secret: str):
+        self._db = db
+        self._secret = secret
+
+    def set_annotation_metadata_from_jwe(self, annotation: Annotation, jwe: str):
+        """
+        Set the decrypted `jwe` dict as the metadata for `annotation`.
+
+        It will create a new AnnotationMetadata row or update the existing one.
```

Comment:
```suggestion
```

I reckon this might be an implementation detail that probably doesn't belong in the docstring: the purpose of having a service on top of the models is so that the service user doesn't have to care what DB tables and rows are involved.